### PR TITLE
[dagster-sdf] Enable Caching & Asset Selection

### PR DESCRIPTION
**Pull request by akbog**
## Summary & Motivation

This PR enables caching simply by upgrading the version of the `sdf-cli` which now outputs cached dependency in std-out in their dependency order.

It also enables asset selection, via sdf's `--targets-only` flag and passing in the fully qualified names of selected assets as targets to the `sdf run` or `sdf test` commands. There are some hoops to jump through with regards to asset checks, but this logic will be cleaned up on the `sdf-cli` in future releases.

Bonus:
1. Duplicate Descriptions (this is fixed with the latest release of the `sdf-cli`)
2. The `--log-form` flag is not necessary with the latest release

## How I Tested These Changes

Tests for caching have been added. Asset selection is covered by existing tests.

### CODE REVIEW

**Comment by cmpadden** [comment]
@akbog, just a heads up that there are a couple of pyright errors:
```
2024-08-21T14:24:27Z] /workdir/python_modules/libraries/dagster-sdf/setup.py:
[2024-08-21T14:24:27Z]   16:11: Expression of type "tuple[str, str, str]" is incompatible with return type "Tuple[str, str]"
[2024-08-21T14:24:27Z]   "tuple[str, str, str]" is incompatible with "Tuple[str, str]"
[2024-08-21T14:24:27Z]     Tuple size mismatch; expected 2 but received 3 (reportReturnType)
[2024-08-21T14:24:27Z]
[2024-08-21T14:24:27Z]   23:0: Expression with type "Tuple[str, str]" cannot be assigned to target tuple
[2024-08-21T14:24:27Z]   Type "Tuple[str, str]" is incompatible with target tuple
[2024-08-21T14:24:27Z]     Tuple size mismatch; expected 3 but received 2 (reportAssignmentType)```

**Comment by akbog** [comment]
> @akbog, just a heads up that there are a couple of pyright errors:
> 
> ```
> 2024-08-21T14:24:27Z] /workdir/python_modules/libraries/dagster-sdf/setup.py:
> [2024-08-21T14:24:27Z]   16:11: Expression of type "tuple[str, str, str]" is incompatible with return type "Tuple[str, str]"
> [2024-08-21T14:24:27Z]   "tuple[str, str, str]" is incompatible with "Tuple[str, str]"
> [2024-08-21T14:24:27Z]     Tuple size mismatch; expected 2 but received 3 (reportReturnType)
> [2024-08-21T14:24:27Z]
> [2024-08-21T14:24:27Z]   23:0: Expression with type "Tuple[str, str]" cannot be assigned to target tuple
> [2024-08-21T14:24:27Z]   Type "Tuple[str, str]" is incompatible with target tuple
> [2024-08-21T14:24:27Z]     Tuple size mismatch; expected 3 but received 2 (reportAssignmentType)
> ```

@cmpadden - should be fixed now!

File: python_modules/libraries/dagster-sdf/dagster_sdf/asset_utils.py [hunk 1]
```diff
@@ -36,7 +36,7 @@ def default_asset_check_key_fn(catalog: str, schema: str, table: str) -> AssetCh
     else:
         asset_key = AssetKey([catalog, schema, table])
     return AssetCheckKey(
-        name=f"{catalog}.{schema}.{table}",
+        name="test",
```
Comment (cmpadden) [inline python_modules/libraries/dagster-sdf/dagster_sdf/asset_utils.py:39]: Is this intentional?

Comment (cmpadden) [reply python_modules/libraries/dagster-sdf/dagster_sdf/asset_utils.py:39]: Nevermind, I see where this is used further down.

Comment (cmpadden) [reply python_modules/libraries/dagster-sdf/dagster_sdf/asset_utils.py:39]: @akbog does this not cause issues in the check having a unique identifier for the table?

Comment (akbog) [reply python_modules/libraries/dagster-sdf/dagster_sdf/asset_utils.py:39]: It does not! There's always a single check per table, that aggregates all checks. This feature in general needs work, but this does not break it!

**Review by cmpadden — APPROVED** [review]

**Pull request merged**